pkg/azuredisk/controllerserver_test.go

```diff
@@ -129,3 +147,128 @@ func TestGetEntriesAndNextToken(t *testing.T) {
 		}
 	}
 }
+
+func TestCreateVolume(t *testing.T) {
+	d := NewFakeDriver()
+
+	tests := []struct {
+		desc            string
+		req             *csi.CreateVolumeRequest
+		expectedResp    *csi.CreateVolumeResponse
+		expectedErrCode codes.Code
+	}{
+		{
+			desc: "fail with no name",
+			req: &csi.CreateVolumeRequest{
+				Name: "",
+			},
+			expectedResp:    nil,
+			expectedErrCode: codes.InvalidArgument,
+		},
+		{
+			desc: "fail with no volume capabilities",
+			req: &csi.CreateVolumeRequest{
+				Name: testVolumeName,
+			},
+			expectedResp:    nil,
+			expectedErrCode: codes.InvalidArgument,
+		},
+		{
+			desc: "fail with the invalid capabilities",
+			req: &csi.CreateVolumeRequest{
+				Name:               testVolumeName,
+				VolumeCapabilities: createVolumeCapabilities(csi.VolumeCapability_AccessMode_UNKNOWN),
+			},
+			expectedResp:    nil,
+			expectedErrCode: codes.InvalidArgument,
+		},
+		{
+			desc: "fail with the invalid requested size",
+			req: &csi.CreateVolumeRequest{
+				Name:               testVolumeName,
+				VolumeCapabilities: stdVolumeCapabilities,
+				CapacityRange: &csi.CapacityRange{
+					RequiredBytes: volumehelper.GiBToBytes(20),
+					LimitBytes:    volumehelper.GiBToBytes(15),
+				},
+			},
+			expectedResp:    nil,
+			expectedErrCode: codes.InvalidArgument,
+		},
+		{
+			desc: "success standard",
+			req: &csi.CreateVolumeRequest{
+				Name:               testVolumeName,
+				VolumeCapabilities: stdVolumeCapabilities,
+				CapacityRange:      stdCapacityRange,
+			},
+			expectedResp: &csi.CreateVolumeResponse{
+				Volume: &csi.Volume{
+					VolumeId:      testVolumeName, //fmt.Sprintf(managedDiskPath, d.cloud.SubscriptionID, d.cloud.ResourceGroup, testVolumeName),
```

Comment:
remove this `//fmt.Sprintf(managedDiskPath, d.cloud.SubscriptionID, d.cloud.ResourceGroup, testVolumeName),`

Reply:
I will remove it in another unit test related pr.